### PR TITLE
Move state-modifying code to QS.update()

### PR DESCRIPTION
**Pull request by lsc36**
Make `QS.make_query()` read-only and ensure that consecutive `make_query()` calls generate results from the same state.

### CODE REVIEW

File: libact/query_strategies/query_by_committee.py [hunk 1]
```diff
@@ -84,6 +83,13 @@ def make_query(self):
                                'resample the bag.')
             student.train(bag)
 
+    def update(self, entry_id, label):
+        self.teach_students()
```
Comment (lsc36) [inline libact/query_strategies/query_by_committee.py]: Should we put `teach_students()` code here instead of calling it?

Comment (yangarbiter) [inline libact/query_strategies/query_by_committee.py]: I think keeping it this way is fine since I also have to call it in __init__.

**Comment by yangarbiter** [comment]
@tungen Would you like to help review this branch so it can be merged?

**Comment by yangarbiter** [comment]
@lsc36 Should we merge now?

**Pull request merged**